src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -175,110 +263,145 @@ private WordEmbeddingsTransform(IHost host, ModelLoadContext ctx, IDataView inpu
-        }
+                Contracts.CheckValue(inputSchema, nameof(inputSchema));
+                Contracts.CheckValue(parent, nameof(parent));
 
-        private ValueGetter<VBuffer<float>> GetGetterVec(IChannel ch, IRow input, int iinfo)
-        {
-            Host.AssertValue(ch);
-            ch.AssertValue(input);
-            ch.Assert(0 <= iinfo && iinfo < Infos.Length);
+                _parent = parent;
+                _inputTypes = new ColumnType[_parent.ColumnPairs.Length];
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    _inputTypes[i] = inputSchema.GetColumnType(ColMapNewToOld[i]);
+                    if (!_inputTypes[i].IsVector || !_inputTypes[i].ItemType.IsText)
+                        throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", _parent.ColumnPairs[i].input, new VectorType(TextType.Instance).ToString(), _inputTypes[i].ToString());
+                }
+
+                _currentVocab = _parent.GetVocabularyDictionary();
+                _outputType = new VectorType(NumberType.R4, 3 * _currentVocab.Dimension);
+            }
 
-            var info = Infos[iinfo];
-            ch.Assert(info.TypeSrc.IsVector);
-            ch.Assert(info.TypeSrc.ItemType.IsText);
+            public override RowMapperColumnInfo[] GetOutputColumns()
+                => _parent.ColumnPairs.Select(x => new RowMapperColumnInfo(x.output, _outputType, null)).ToArray();
 
-            var srcGetter = input.GetGetter<VBuffer<DvText>>(info.Source);
-            var src = default(VBuffer<DvText>);
-            int dimension = _currentVocab.Dimension;
-            float[] wordVector = new float[_currentVocab.Dimension];
+            protected override Delegate MakeGetter(IRow input, int iinfo, out Action disposer)
+            {
+                Contracts.AssertValue(input);
+                Contracts.Assert(0 <= iinfo && iinfo < _parent.ColumnPairs.Length);
+                disposer = null;
 
-            return
-                (ref VBuffer<float> dst) =>
+                var colType = _inputTypes[iinfo];
+                if (!colType.IsVector)
                 {
-                    int deno = 0;
-                    srcGetter(ref src);
-                    var values = dst.Values;
-                    if (Utils.Size(values) != 3 * dimension)
-                        values = new float[3 * dimension];
-                    int offset = 2 * dimension;
-                    for (int i = 0; i < dimension; i++)
-                    {
-                        values[i] = float.MaxValue;
-                        values[i + dimension] = 0;
-                        values[i + offset] = float.MinValue;
-                    }
-                    for (int word = 0; word < src.Count; word++)
+                    throw Host.Except(_parent.ColumnPairs[iinfo].input,
+                        "Text input given, expects a text vector");
+                }
```

Comment:
Why you run this check here? why not in constructor?

Reply:
I think its not need here anymore as constructor is checking it already.

---
In reply to: [218256497](https://github.com/dotnet/machinelearning/pull/928#discussion_r218256497) [](ancestors = 218256497)